crates/services/txpool_v2/src/collision_manager/basic.rs

```diff
@@ -236,4 +239,71 @@ where
             };
         }
     }
+
+    #[cfg(test)]
+    fn assert_integrity(&self, expected_txs: &[ArcPoolTx]) {
+        use std::ops::Deref;
+
+        let mut message_spenders = HashMap::new();
+        let mut coins_spenders = BTreeMap::new();
+        let mut contracts_creators = HashMap::new();
+        let mut blobs_users = HashMap::new();
+        for tx in expected_txs {
+            if let PoolTransaction::Blob(checked_tx, _) = tx.deref() {
+                let blob_id = checked_tx.transaction().blob_id();
+                blobs_users.insert(*blob_id, tx.id());
+            }
+            for input in tx.inputs() {
+                match input {
+                    Input::CoinSigned(CoinSigned { utxo_id, .. })
+                    | Input::CoinPredicate(CoinPredicate { utxo_id, .. }) => {
+                        coins_spenders.insert(*utxo_id, tx.id());
+                    }
+                    Input::MessageCoinSigned(MessageCoinSigned { nonce, .. })
+                    | Input::MessageCoinPredicate(MessageCoinPredicate {
+                        nonce, ..
+                    })
+                    | Input::MessageDataSigned(MessageDataSigned { nonce, .. })
+                    | Input::MessageDataPredicate(MessageDataPredicate {
+                        nonce, ..
+                    }) => {
+                        message_spenders.insert(*nonce, tx.id());
+                    }
+                    _ => {}
+                }
+            }
+            for output in tx.outputs() {
+                match output {
+                    Output::Coin { .. }
+                    | Output::Change { .. }
+                    | Output::Variable { .. }
+                    | Output::Contract(_) => {}
+                    Output::ContractCreated { contract_id, .. } => {
+                        contracts_creators.insert(*contract_id, tx.id());
+                    }
+                }
+            }
```

Comment:
this can be shortened a bit: 
```suggestion
            for output in tx.outputs() {
                if let Output::ContractCreated { contract_id, .. } = output {
                    contracts_creators.insert(*contract_id, tx.id());
                }
            }
```